LemonadeHomeTransition/ScrollableViewController.swift

```diff
@@ -0,0 +1,181 @@
+//
+//  ScrollableViewController.swift
+//  LemonadeHomeTransition
+//
+//  Created by Dang Quoc Huy on 4/21/17.
+//  Copyright © 2017 Dang Quoc Huy. All rights reserved.
+//
+
+import UIKit
+
+protocol UIAnimateViewController {
+  var isFrom: Bool! { get set }
+  var index: Int! { get set }
+  
+  func setupInterpolationsFrom()
+  func animateFrom(progress: CGFloat)
+  func invalidateFrom()
+  
+  func setupInterpolationsTo()
+  func animateTo(progress: CGFloat)
+  func invalidateTo()
+}
+
+extension UIAnimateViewController {
+  
+  final func setupInterpolations(isFrom: Bool) {
+    isFrom ? setupInterpolationsFrom() : setupInterpolationsTo()
+  }
+  
+  final mutating func animate(progress: CGFloat, isFrom: Bool) {
+    guard progress >= 0 && progress <= 1 else { return }
+    
+    if self.isFrom == nil {
```

Comment:
one thing that might really help the readability of this code is to make silly little methods for all this == 1 stuff or == nil stuff.

so for example
```
if !self.isFromSet() {
   ....
``` 

(where isFromSet is just a silly method where == nil)

same goes for `isFrom != self.isFrom` could be renamed to a function called `isNewFrom` or something

and below, on == 1 could just be `self.isTransitionComplete)`